src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
+export function groupByKey<T>(items, key): GroupsMap<T>[] {
   // write code here;
+  return items.reduce((acc, item) => {
+    const add = items.filter((itemForFilter) => (
+      item[key] === itemForFilter[key]
+    ));
+
+    return acc[item[key]]
+      ? acc
+      : {
+        ...acc,
+        [item[key]]: add,
+      };
+  }, {});
```

Comment:
As I see, you are also a fan of the reduce method, like me ;0)
here is another possible solution:
`
export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
  return items.reduce((acc: GroupsMap<T>, item) => {
    const groupBy = `${item[key]}`;

    acc[groupBy] = (acc[groupBy] || []).concat(item);

    return acc;
  }, {});
}
`